Tests/AlicerceTests/Analytics/AnalyticsTestCase.swift

```diff
@@ -0,0 +1,154 @@
+//
+//  AnalyticsTestCase.swift
+//  Alicerce
+//
+//  Created by Luís Portela on 27/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import XCTest
+
+@testable import Alicerce
+
+final class AnalyticsTestCase: XCTestCase {
+    
+    final class TestableTracker: AnalyticsTracker {
+        private let expectation: XCTestExpectation
+        
+        var mockPage: Analytics.Page
+        var mockEvent: Analytics.Event
+        
+        init(expectation: XCTestExpectation,
+             page: Analytics.Page = Analytics.Page(name: "page", parameters: nil),
+             event: Analytics.Event = Analytics.Event(name: "event", parameters: nil)) {
+            
+            self.expectation = expectation
+            self.mockPage = page
+            self.mockEvent = event
+        }
+        
+        func track(page: Analytics.Page) {
+            XCTAssertEqual(page, mockPage)
+            
+            expectation.fulfill()
+        }
+        
+        func track(event: Analytics.Event) {
+            XCTAssertEqual(event, mockEvent)
+            
+            expectation.fulfill()
+        }
+    }
+    
+    func testAddParameters_WithACleanListOfParameters_ItShouldAppendTheNewParameters() {
+        let analytics = Analytics()
+        
+        XCTAssertNil(analytics.parameters)
+        
+        analytics.add(parameters: ["1" : "1"])
+        
+        XCTAssertNotNil(analytics.parameters)
+        XCTAssertNotNil(analytics.parameters?.first)
+        XCTAssertEqual(analytics.parameters?.count, 1)
+        let firstParameter = analytics.parameters?.first
+        XCTAssertEqual(firstParameter?.key, "1")
+        XCTAssertEqual(firstParameter?.value as! String, "1")
+    }
+    
+    func testAddParameters_WithADirtyList_ItShouldAppendTheNewParameters() {
+        let analyticsConfiguration = Analytics.Configuration(extraParameters: [
+            "1" : "1"
+        ])
+        
+        let analytics = Analytics(configuration: analyticsConfiguration)
+        
+        XCTAssertNotNil(analytics.parameters)
+        XCTAssertEqual(analytics.parameters?.count, 1)
+        
+        analytics.add(parameters: ["2" : "2"])
+        
+        XCTAssertEqual(analytics.parameters?.count, 2)
+        XCTAssertEqual(analytics.parameters!["1"] as! String, "1")
+        XCTAssertEqual(analytics.parameters!["2"] as! String, "2")
+    }
+    
+    func testAddTracker_WhenATrackerIsAdded_ItShouldSendEventsToThatTracker() {
+        let expectation = self.expectation(description: "Track appended tracker")
+
+        let page = Analytics.Page(name: "trackPage", parameters: nil)
+        let testableTracker = TestableTracker(expectation: expectation, page: page)
+        
+        let analytics = Analytics()
+        analytics.add(tracker: testableTracker)
+        
+        analytics.track(page: page)
+        
+        waitForExpectations(timeout: 30)
```

Comment:
Why not create a constant with this timeout? That way it's easier to change in the future